plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClientModule.java

```diff
@@ -53,33 +54,75 @@ public void configure(Binder binder)
     }
 
     @Provides
     @Singleton
     @ForBaseJdbc
-    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OracleConfig oracleConfig, OpenTelemetry openTelemetry)
-            throws SQLException
+    public ConnectionFactory connectionFactory(
+            OracleConfig oracleConfig,
+            OraclePoolConnectionFactory oraclePoolConnectionFactory,
+            RetryingConnectionFactory retryingConnectionFactory)
     {
-        Properties connectionProperties = new Properties();
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_INCLUDE_SYNONYMS, String.valueOf(oracleConfig.isSynonymsEnabled()));
-        connectionProperties.setProperty(OracleConnection.CONNECTION_PROPERTY_REPORT_REMARKS, String.valueOf(oracleConfig.isRemarksReportingEnabled()));
-
         if (oracleConfig.isConnectionPoolEnabled()) {
-            return new OraclePoolConnectionFactory(
-                    config.getConnectionUrl(),
-                    connectionProperties,
-                    credentialProvider,
-                    oracleConfig.getConnectionPoolMinSize(),
-                    oracleConfig.getConnectionPoolMaxSize(),
-                    oracleConfig.getInactiveConnectionTimeout());
+            return oraclePoolConnectionFactory;
+        }
+        else {
+            return retryingConnectionFactory;
         }
+    }
 
-        return new RetryingConnectionFactory(new DriverConnectionFactory(
-                new OracleDriver(),
+    @Provides
+    @Singleton
+    public OraclePoolConnectionFactory getOraclePoolConnectionFactory(
+            BaseJdbcConfig config,
+            CredentialProvider credentialProvider,
+            OracleConfig oracleConfig,
+            @Named("oracleConnectionProperties") Properties connectionProperties)
+            throws SQLException
+    {
+        return new OraclePoolConnectionFactory(
                 config.getConnectionUrl(),
                 connectionProperties,
                 credentialProvider,
-                openTelemetry));
+                oracleConfig.getConnectionPoolMinSize(),
+                oracleConfig.getConnectionPoolMaxSize(),
+                oracleConfig.getInactiveConnectionTimeout());
+    }
+
+    @Provides
+    @Singleton
+    @Named("retryingDelegate")
+    public ConnectionFactory getDriverConnectionFactory(
+            OracleDriver oracleDriver,
+            BaseJdbcConfig config,
+            CredentialProvider credentialProvider,
+            OpenTelemetry openTelemetry,
+            @Named("oracleConnectionProperties") Properties connectionProperties)
+    {
+        return new DriverConnectionFactory(
+                oracleDriver,
+                config.getConnectionUrl(),
+                connectionProperties,
+                credentialProvider,
+                openTelemetry);
+    }
+
+    @Provides
+    @Singleton
+    @Named("oracleConnectionProperties")
```

Comment:
Do we have other `Properties` in context?

Reply:
No. Initially I labelled them in case there was a clash with some other Properties elsewhere. Now, I introduced PrivateModule that hides this from the rest of there world. This allowed me to remove the label.